### ISSUE TITLE
LSP picks up `@types/*` package only with "manual" node_modules dir handling

### ISSUE THREAD

**Issue by bartlomieju** [denoland/deno#27215]
Version: Deno 2.1.2

When working on a project with the Deno LSP, Deno it not picking up `@types/*` package automatically, unless `nodeModulesDir` setting is set to `"manual"`. That means in situation when `nodeModulesDir` is explicitly set to "auto", the typings for npm packages are not found automatically.

With no `nodeModulesDir` or with `nodeModulesDir` set to `"manual":
![Screenshot 2024-12-03 at 15 44 25](https://github.com/user-attachments/assets/102bfb06-8b5c-42a7-a069-a736bf308892)

With `nodeModulesDir` set to `"auto"`:
![Screenshot 2024-12-03 at 15 44 53](https://github.com/user-attachments/assets/fc20cd39-058a-4da8-9db7-b5d32cdf733b)

In other words the LSP should look inside `node_modules` directory for `@types/*` packages and always use them if they are available.
